pkg/resourcecollector/resourcecollector.go

```diff
@@ -522,6 +524,52 @@ func (r *ResourceCollector) GetResources(
 	return allObjects, pvcsWithOwnerReference, nil
 }
 
+// gatherResourceInChunks() collects all the resources present per ns or resource types.
+// It does it in batch of 500 unless specified by user via a config-map. This function
+// primarily helps in avoiding server timeout error when the number of resource or size
+// of it large.
+func gatherResourceInChunks(dynamicClient dynamic.ResourceInterface, opts Options, selectors string) (*unstructured.UnstructuredList, error) {
+	var err error
+	fn := "gatherResourceInChunks"
+	// Read configMap for any user provided resource count limit to be provided to List call.
+	resourceCountLimit := opts.ResourceCountLimit
+	resourceCountLimitString, err := k8sutils.GetConfigValue(k8sutils.StorkControllerConfigMapName, metav1.NamespaceSystem, k8sutils.ResourceCountLimitKeyName)
```

Comment:
I think reading the ConfigMap should be outside of the resource collector. This might not be applicable to other module, who callout.

Reply:
Sure.. else we will overwrite the value provided by UI. Specifically when they also invoke this independently and expects to use environment variable provided value. Will change the code.